packages/ember-runtime/lib/system/core_object.js

```diff
@@ -180,11 +180,7 @@ function makeCtor(base) {
               let baseValue = self[keyName];
 
               if (hasConcatenatedProps && concatenatedProperties.indexOf(keyName) > -1) {
-                if (baseValue) {
-                  value = makeArray(baseValue).concat(value);
-                } else {
-                  value = makeArray(value);
-                }
+                value = makeArray(baseValue).concat(makeArray(value));
```

Comment:
The reason for the previous implementation is that we specifically do not want to create the extra array for every `Ember.Object.create()`.

Reply:
but that fixes inconsistency shown in test